### PR TITLE
Rarick docker updates

### PR DESCRIPTION
**Pull request by rarick**
Added a development dockerfile. In the future, a `/partyball` directory will be mounted containing source code that we edit. The `gfootball` directory should be largely be treated like a base module, and should not edit files in there unless required.

### CODE REVIEW

File: docker/dev.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+RUN apt-get update && \
+      apt-get --no-install-recommends install -yq \
+      git cmake build-essential \
+      libgl1-mesa-dev libsdl2-dev \
+      libsdl2-image-dev libsdl2-ttf-dev libsdl2-gfx-dev libboost-all-dev \
+      libdirectfb-dev libst-dev mesa-utils xvfb x11vnc \
+      libsdl-sge-dev python3-pip && \
+      pip3 install --upgrade pip setuptools psutil && \
+      pip3 install /tmp/gfootball && \
```
Comment (AdrianaVega97) [inline docker/dev.Dockerfile:15]: What is the tmp directory ?

Comment (rarick) [reply docker/dev.Dockerfile:15]: https://www.howtogeek.com/117435/htg-explains-the-linux-directory-structure-explained/

Comment (rarick) [reply docker/dev.Dockerfile:15]: Traditionally in linux, `/tmp` is a good place to put temporary items. It's also normally wiped out on reboot.

Here I delete the files after install, so they're put into `/tmp` to indicate they won't be sticking around.

**Comment by GeneraleLee** [comment]
dev Dockerfile works like a charm for me

**Review by GeneraleLee — APPROVED** [review]
Works perfectly for me. I have run "xhost +"local:docker@" before running the image. It is probably important for rendering.

**Pull request merged**